jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/MavenProjectProperties.java

```diff
@@ -200,23 +184,55 @@ static int getVersionFromString(String versionString) {
   private final MavenProject project;
   private final SingleThreadedExecutor singleThreadedExecutor = new SingleThreadedExecutor();
   private final EventHandlers eventHandlers;
-  private final JavaLayerConfigurations javaLayerConfigurations;
+  private final AbsoluteUnixPath appRoot;
 
   @VisibleForTesting
   MavenProjectProperties(
-      MavenProject project,
-      MavenSession session,
-      Log log,
-      JavaLayerConfigurations javaLayerConfigurations) {
+      MavenProject project, MavenSession session, Log log, AbsoluteUnixPath appRoot) {
     this.project = project;
-    this.javaLayerConfigurations = javaLayerConfigurations;
-
+    this.appRoot = appRoot;
     eventHandlers = makeEventHandlers(session, log, singleThreadedExecutor);
   }
 
   @Override
-  public JavaLayerConfigurations getJavaLayerConfigurations() {
-    return javaLayerConfigurations;
+  public JibContainerBuilder getContainerBuilderWithLayers(RegistryImage baseImage)
+      throws IOException {
+    try {
+      if (MojoCommon.isWarProject(project)) {
+        Path explodedWarPath =
+            Paths.get(project.getBuild().getDirectory()).resolve(project.getBuild().getFinalName());
+        return JavaContainerBuilderHelper.fromExplodedWar(baseImage, explodedWarPath, appRoot);
+      }
+
+      Path classesOutputDirectory = Paths.get(project.getBuild().getOutputDirectory());
+      Predicate<Path> isClassFile = path -> path.getFileName().toString().endsWith(".class");
+
+      // Add dependencies, resources, and classes
+      return JavaContainerBuilder.from(baseImage)
+          .setAppRoot(appRoot)
+          .addResources(classesOutputDirectory, isClassFile.negate())
+          .addClasses(classesOutputDirectory, isClassFile)
+          .addDependencies(
+              project
+                  .getArtifacts()
+                  .stream()
+                  .map(Artifact::getFile)
+                  .map(File::toPath)
+                  .collect(Collectors.toList()))
+          .toContainerBuilder();
+
+    } catch (IOException ex) {
+      throw new IOException(
+          "Obtaining project build output files failed; make sure you have compiled your project "
+              + "before trying to build the image. (Did you accidentally run \"mvn clean "
+              + "jib:build\" instead of \"mvn clean compile jib:build\"?)",
+          ex);
+    }
+  }
+
+  @Override
+  public ImmutableList<Path> getClassFiles() throws IOException {
+    return new DirectoryWalker(Paths.get(project.getBuild().getOutputDirectory())).walk().asList();
```

Comment:
This will include non-`.class` files (e.g., resource files). Will this be OK?

Reply:
For our use case, it should be, since the main class finder already filters for the ".class" extension. But I suppose the method name is misleading.